src/lib.rs

```diff
@@ -837,16 +837,22 @@ impl Config {
         let mut use_jobserver = false;
         if fs::metadata(build.join("Makefile")).is_ok() {
             match env::var_os("CARGO_MAKEFLAGS") {
-                // Only do this on non-windows and non-bsd
-                // On Windows, we could be invoking make instead of
-                // mingw32-make which doesn't work with our jobserver
-                // bsdmake also does not work with our job server
+                // Only do this on non-windows, non-bsd, and non-macos (unless a named pipe
+                // jobserver is available)
+                // * On Windows, we could be invoking make instead of
+                //   mingw32-make which doesn't work with our jobserver
+                // * bsdmake also does not work with our job server
+                // * On macOS, CMake blocks propagation of the jobserver's file descriptors to make
+                //   However, if the jobserver is based on a named pipe, this will be available to
+                //   the build.
                 Some(ref makeflags)
                     if !(cfg!(windows)
                         || cfg!(target_os = "openbsd")
                         || cfg!(target_os = "netbsd")
                         || cfg!(target_os = "freebsd")
-                        || cfg!(target_os = "dragonfly")) =>
+                        || cfg!(target_os = "dragonfly")
+                        || (cfg!(target_os = "macos")
```

Comment:
Does this apply to other Apple platforms? If so, `target_family = "apple"` may be more applicable.

Reply:
I'm open to whichever option you prefer. The CMake/libuv fd-closing behavior is gated on `defined(__APPLE__) && MAC_OS_X_VERSION_MIN_REQUIRED >= 1080`. See https://github.com/Kitware/CMake/blob/master/Utilities/cmlibuv/src/unix/process.c#L39-L44 for reference.

Reply:
Just go with `target_os = "macos"`, that's clearer here

Reply:
sounds good thanks